plaso/parsers/locate.py

```diff
@@ -0,0 +1,191 @@
+
+          if db_type == self._DBE_END:
+            break
+          self._ParseCString()
+      except struct_error:
+        raise errors.UnableToParseFile('Error in header')
+
+  def Open(self, file_object):
+    """Opens a Locate database file.
+
+    Args:
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      IOError: if the file-like object cannot be read.
+      OSError: if the file-like object cannot be read.
+      ValueError: if the file-like object is missing.
+      UnableToParseFile: if the file cannot be parsed
+    """
+    if not file_object:
+      raise ValueError('Missing file object.')
+
+    self._file_object = file_object
+    self._file_object.seek(0, os.SEEK_SET)
+
+    header_bytes = self._file_object.read(16)
+    if len(header_bytes) != 16:
+      raise errors.UnableToParseFile('Unable to read the file header')
+
+    try:
+      (magic, conf_size, _version, _check_visibility, _pad) = \
+        struct_unpack(self._DB_HEADER, header_bytes)
+      if magic != self.DB_MAGIC:
+        raise errors.UnableToParseFile('Invalid file magic')
+
+      self.root_path = self._ParseCString()
+
+      # skip configuration section for now
+      self._file_object.seek(conf_size, os.SEEK_CUR)
+    except struct_error:
+      raise errors.UnableToParseFile('Unable to unpack the file header')
+
+
+class LocateDatabaseParser(interface.FileObjectParser):
+  """Parser for locate/updatedb database files"""
+
+  NAME = 'locate_database'
+  DATA_FORMAT = 'Locate Database file'
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification
+
+    Returns:
+      FormatSpecification: format spefication.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(LocateDatabaseFile.DB_MAGIC, offset=0)
+    return format_specification
+
+  #pylint: disable=unused-argument
+  def ParseFileObject(self, parser_mediator, file_object, **kwargs):
+    """Parses a locate/updatedb database file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): parser mediator.
+      file_object (dfvfs.FileIO): file-like object to be parsed.
+
+    Raises:
+      UnableToParseFile: when the file cannot be parsed, this will signal
+        the event extractor to apply other parsers.
+    """
+    locate_file = LocateDatabaseFile()
+
+    try:
+      locate_file.Open(file_object)
+
+      for (folder_path, folder_timestamp) in locate_file.ParsePaths():
+        event_data = LocateDatabaseEvent()
+        event_data.folder_path = folder_path
+
+        event = time_events.DateTimeValuesEvent(folder_timestamp,
+          definitions.TIME_DESCRIPTION_MODIFICATION)
+        parser_mediator.ProduceEventWithEventData(event, event_data)
+    except Exception as exception:  # pylint: disable=broad-except
+      raise errors.UnableToParseFile(
+        'unable to parse locate database with error: {exception}'.format(
```

Comment:
To have the style match the other parsers

`{exception}` => `{0!s}`
`exception=exception` => `exception`

Reply:
Updated style throughout file